src/control/server/ctl_storage_rpc.go

```diff
@@ -366,18 +192,35 @@ func (c *ControlService) doFormat(srv *IOServerInstance, reformat bool, resp *ct
 // Send response containing multiple results of format operations on scm mounts
 // and nvme controllers.
 func (c *ControlService) StorageFormat(ctx context.Context, req *ctlpb.StorageFormatReq) (*ctlpb.StorageFormatResp, error) {
-	resp := new(ctlpb.StorageFormatResp)
-	resp.Mrets = proto.ScmMountResults{}
-	resp.Crets = proto.NvmeControllerResults{}
+	var resp *ctlpb.StorageFormatResp
+	respChan := make(chan *ctlpb.StorageFormatResp, len(c.harness.Instances()))
```

Comment:
I was kind of hoping to further decouple the core storage formatting logic from the details of handling RPC requests, but this is a reasonable step in that direction.